pydicom/util/codify.py

```diff
@@ -108,53 +119,67 @@ def code_dataelem(dataelem,
     valuerep = repr(dataelem.value)
 
     if exclude_size:
-        if (dataelem.VR in byte_VRs and
-                len(dataelem.value) > exclude_size):
-            valuerep = (
-                "# XXX Array of %d bytes excluded" % len(dataelem.value))
+        if (
+            dataelem.VR in BINARY_VR_VALUES
+            and not isinstance(dataelem.value, (int, float))
+            and len(dataelem.value) > exclude_size
+        ):
+            valuerep = f"# XXX Array of {len(dataelem.value)} bytes excluded"
 
     if have_keyword:
-        format_str = "{ds_name}.{keyword} = {valuerep}"
-        line = format_str.format(
-            ds_name=dataset_name, keyword=keyword, valuerep=valuerep)
+        line = f"{dataset_name}.{keyword} = {valuerep}"
     else:
-        format_str = "{ds_name}.add_new({tag}, '{VR}', {valuerep})"
-        line = format_str.format(
-            ds_name=dataset_name,
-            tag=tag_repr(dataelem.tag),
-            VR=dataelem.VR,
-            valuerep=valuerep)
+        tag = tag_repr(dataelem.tag)
+        VR = dataelem.VR
+        line = f"{dataset_name}.add_new({tag}, '{VR}', {valuerep})"
```

Comment:
I may have mentioned that I like f-strings...

Reply:
Yeah, I love them too...  so much easier to read and understand instantly.